solutions/devsprint-pedro-menezes-1/DeliveryAppChallenge/Modules/Product/Scenes/BrunaTokie/BTResetPasswordViewController.swift

```diff
@@ -83,24 +51,82 @@ class BTResetPasswordViewController: UIViewController {
         newVc.modalPresentationStyle = .fullScreen
         present(newVc, animated: true)
     }
+
+    private func isRecoveryEmail() -> Bool {
+        if recoveryEmail {
+            dismiss(animated: true)
+            return false
+        } else {
+            return true
+        }
+    }
+
+    private func checkConectionAvaiability() {
+        if !ConnectivityManager.shared.isConnected {
+            Globals.showNoInternetCOnnection(controller: self)
+        } else {
+            recoverPasswordRequest()
+        }
+    }
+
+    private func recoverPasswordRequest() {
+        let emailUser = emailTextfield.text!.trimmingCharacters(in: .whitespaces)
+
+        let parameters = [
+            "email": emailUser
+        ]
+        recoverPasswordData(parameters: parameters)
+    }
+
+    private func recoverPasswordData(parameters: [String: String]) {
+        BadNetworkLayer.shared.resetPassword(self, parameters: parameters) { (success) in
+            if success {
+                self.newPasswordCreatedSuccess()
+            } else {
+                self.createErrorAlert()
+            }
+        }
+    }
+
+    private func createErrorAlert() {
+        let alertController = UIAlertController(title: "Ops..", message: "Algo de errado aconteceu. Tente novamente mais tarde.", preferredStyle: .alert)
+        let action = UIAlertAction(title: "OK", style: .default)
+        alertController.addAction(action)
+        self.present(alertController, animated: true)
+    }
+
+    private func newPasswordCreatedSuccess() {
+        self.recoveryEmail = true
+        self.emailTextfield.isHidden = true
+        self.textLabel.isHidden = true
+        self.viewSuccess.isHidden = false
+        self.emailLabel.text = self.emailTextfield.text?.trimmingCharacters(in: .whitespaces)
+        self.recoverPasswordButton.titleLabel?.text = "REENVIAR E-MAIL"
+        self.recoverPasswordButton.setTitle("Voltar", for: .normal)
+    }
     
-    func validateForm() -> Bool {
-        let status = emailTextfield.text!.isEmpty ||
+    private func isValidatedForm() -> Bool {
+        let isValid = !(emailTextfield.text!.isEmpty ||
             !emailTextfield.text!.contains(".") ||
             !emailTextfield.text!.contains("@") ||
-            emailTextfield.text!.count <= 5
-        
-        if status {
+            emailTextfield.text!.count <= 5)
+
+        if !isValid {
+            formNotValidated(isValid: false)
+        }
+        return isValid
+    }
+
+    private func formNotValidated(isValid: Bool) {
+        if !isValid {
```

Comment:
se essa funcao só tem ação quando o isValid é false, então vc n precisa passar o parâmetro
e aonde vc chama essa funcao, vc só chama se realmente o formulário nao for valido

Reply:
Acabei criando algo muito over... kkk